### PR TITLE
Proofpoint TAP Playbooks

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes
  - Add Proofpoint TAP Custom Logic App Connector and Playbooks

### CODE REVIEW

**Review by sarah-yo — CHANGES_REQUESTED** [review]
- Please check spelling, grammar and correct capitalization in all readme files.
- Please expand the instructions for "b. Configurations in Sentinel" in each readme

**Comment by lior-tamir** [comment]
I have sent feedback over email to the contributors

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,280 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
```
Comment (v-admahe) [inline Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/azuredeploy.json:3]: Metadata parameters is missing.

Thanks,
Aditi Maheshwari

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddInfoToIncident/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,318 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
```
Comment (v-admahe) [inline Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddInfoToIncident/azuredeploy.json:3]: Metadata parameters is missing.

Thanks,
Aditi Maheshwari

File: Playbooks/ProofpointTAP/ProofpointTAPConnector/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,905 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
```
Comment (v-admahe) [inline Playbooks/ProofpointTAP/ProofpointTAPConnector/azuredeploy.json:3]: Metadata parameters is missing.

Thanks,
Aditi Maheshwari

File: Playbooks/ProofpointTAP/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
```
Comment (v-admahe) [inline Playbooks/ProofpointTAP/azuredeploy.json:3]: Metadata parameters is missing.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-jayakal** [comment]
@vu-socprime , @sarah-yo , @lior-tamir & @sreedharande - As we already have a solution for Proofpoint TAP, these playbooks will be packaged as part of the solution. As part of that, these files will be moved to solution folder once this is approved. Thanks.

**Comment by vu-socprime** [comment]
> I have sent feedback over email to the contributors

@lior-tamir Thanks! Playbooks have been updated according to the email.

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-CheckAccountInVAP/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for Proofpoint TAP connector API Connection (For authorizing, the Service Principal and the secret need to be provided.)
+#### b. Configurations in Sentinel
+1. In Azure sentinel analytical rules should be configured to trigger an incident with risky user account. In the *Entity maping* section of the analytics rule creation workflow, user's email should be mapped to **FullName** identitfier of the **Account** entity type. Check the [documentation](https://docs.microsoft.com/en-us/azure/sentinel/map-data-fields-to-entities) to learn more about mapping entities.
```
Comment (sarah-yo) [inline Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-CheckAccountInVAP/readme.md:31]: remove "en-us" from URL

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/readme.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+#### a. Authorize connections
+Once deployment is complete, you will need to authorize each connection.
+1.	Click the Azure Sentinel connection resource
+2.	Click edit API connection
+3.	Click Authorize
+4.	Sign in
+5.	Click Save
+6.	Repeat steps for Proofpoint TAP connector API Connection (For authorizing, the Service Principal and the secret need to be provided.)
+#### b. Configurations in Sentinel
+1. In Azure sentinel analytical rules should be configured to trigger an incident. An incident should have *campaignId* custom entity (obtained from *campaignId_s* field in ProofpointTAP logs). Check the [documentation](https://docs.microsoft.com/en-us/azure/sentinel/surface-custom-details-in-alerts) to learn more about adding custom entities to incidents.
```
Comment (sarah-yo) [inline Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/readme.md:30]: remove "en-us" from URL

**Review by sarah-yo — CHANGES_REQUESTED** [review]
Please remove "en-us" from URLs as marked in the readmes

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-CheckAccountInVAP/readme.md (hunk 5 above)
Comment (vu-socprime) [reply Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-CheckAccountInVAP/readme.md]: updated

File: Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/readme.md (hunk 6 above)
Comment (vu-socprime) [reply Playbooks/ProofpointTAP/Playbooks/ProofpointTAP-AddForensicsInfoToIncident/readme.md]: updated

**Comment by vu-socprime** [comment]
> * Please check spelling, grammar and correct capitalization in all readme files.
> * Please expand the instructions for "b. Configurations in Sentinel" in each readme

@sarah-yo Done

**Comment by v-jayakal** [comment]
@vu-socprime , can you please move these files into solutions folder of Cisco Umbrella? Path for the Cisco Umbrella solution is https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/CiscoUmbrella. Please add a folder called Playbooks and move these files. You can refer other solutions with playbooks (ex: Cisco ISE) for reference.

**Comment by v-jayakal** [comment]
@sarah-yo , @lior-tamir - Can you please approve the PR? thanks.

**Comment by v-jayakal** [comment]
@vu-socprime , Can you please move these files into Proofpoint TAP solution folder? Thanks.

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**